backend/wmg/pipeline/summary_cubes/calculate_markers.py

```diff
@@ -291,130 +249,23 @@ def _calculate_true_n_cells(n_cells, genes, dataset_to_gene_ids, keep_dataset_id
 
     # get the tuples of filter values, excluding dataset IDs if they are not specified
     # as a filter by the user.
-    if keep_dataset_ids:
-        groups = list(n_cells.index)
-        level_names = tuple(n_cells.index.names)
-    else:
-        groups = list(zip(*[n_cells.index.get_level_values(i) for i in range(1, len(n_cells.index[0]))]))
-        level_names = tuple(n_cells.index.names[1:])
+    cell_types = n_cells.index.get_level_values("cell_type_ontology_term_id")
 
     # sum up the cell count arrays across duplicate groups
     # (groups can be duplicate after excluding dataset_ids)
     t_n_cells_sum = {}
-    for i, k in enumerate(groups):
+    for i, k in enumerate(cell_types):
         summer = t_n_cells_sum.get(k, np.zeros(t_n_cells.shape[1]))
         summer += t_n_cells[i]
         t_n_cells_sum[k] = summer
 
-    unique_groups = list(t_n_cells_sum.keys())
     n_cells_array = np.vstack(list(t_n_cells_sum.values()))
-    if "cell_type_ontology_term_id" in level_names:
-        cell_type_index = level_names.index("cell_type_ontology_term_id")
-        cell_types = [group[cell_type_index] for group in unique_groups]
-        n_cells_array = rollup_across_cell_type_descendants_array(n_cells_array, cell_types)
 
-    index = pd.Index(unique_groups, name=level_names)
+    n_cells_array = rollup_across_cell_type_descendants_array(n_cells_array, cell_types)
+    index = pd.Index(cell_types, name="cell_type_ontology_term_id")
     return n_cells_array, index
 
 
-def _prepare_indices_and_metrics(target_filters, context_filters, corpus=None):
```

Comment:
this no longer needs to be a separate function as it is not being reused anywhere (previously it was being used in binomial test)